v2/pkg/engine/resolve/inputtemplate.go

```diff
@@ -113,6 +114,9 @@ func (i *InputTemplate) renderObjectVariable(ctx context.Context, variables []by
 		_, _ = preparedInput.Write(literal.NULL)
 		return nil
 	}
+
+	segment.VariableValueType = valueType
```

Comment:
The template is used for concurrent requests. This is a race.
We should never write to a template during resolving.

Reply:
Good catch. I moved the logic to filter.

Reply:
How can we prevent such situations? I suggest to document those constraints and also add tests in the future.

Reply:
You could add a test that will trigger the race.